File: app/src/main/java/co/fddittmar/planbuilder/presenter/NewProgramPresenter.java
package co.fddittmar.planbuilder.presenter;

import java.util.List;

import co.fddittmar.planbuilder.contracts.NewProgramContract;
import co.fddittmar.planbuilder.data.ProgramRepository;
import co.fddittmar.planbuilder.data.model.Exercise;

/**
 * Presenter for the MainActivity. It takes a reference to a View and a Repository.
 */

public class NewProgramPresenter implements NewProgramContract.Actions {

    NewProgramContract.View view;
    ProgramRepository repository;

    public NewProgramPresenter(NewProgramContract.View view, ProgramRepository repository) {
        this.view = view;
        this.repository = repository;
    }

    @Override
    public void saveProgram(String title, String startDate, String endDate, List<Exercise> exercises) {
        repository.saveProgram(title, startDate, endDate, exercises);
        view.returnToMainActivity();
    }

    @Override
    public void onAddNewExercise() {
        view.setupNewExerciseDialog();
    }
}

File: app/src/main/java/co/fddittmar/planbuilder/presenter/NewProgramPresenter.kt
package co.fddittmar.planbuilder.presenter

import co.fddittmar.planbuilder.contracts.NewProgramContract
import co.fddittmar.planbuilder.data.ProgramRepository
import co.fddittmar.planbuilder.data.model.Exercise

/**
 * Presenter for the MainActivity. It takes a reference to a View and a Repository.
 */

class NewProgramPresenter(internal var view: NewProgramContract.View, internal var repository: ProgramRepository) : NewProgramContract.Actions {

    override fun saveProgram(title: String, startDate: String, endDate: String, exercises: List<Exercise>) {
        repository.saveProgram(title, startDate, endDate, exercises)
        view.returnToMainActivity()
    }

    override fun onAddNewExercise() {
        view.setupNewExerciseDialog()
    }
}

